heim-process/src/sys/linux/process/procfs/env.rs

```diff
@@ -7,7 +7,9 @@ use crate::{Pid, ProcessError, ProcessResult};
 pub use crate::sys::unix::{Environment, IntoEnvironmentIter};
 
 pub async fn environment(pid: Pid) -> ProcessResult<Environment> {
-    let path = format!("/proc/{}/environ", pid);
+    let path = rt::linux::procfs_root()
+        .join(pid.to_string())
+        .join("environ");
```

Comment:
This part become really repetative: "procfs root" + pid + "filename", which is probably also does two memory allocations: first one for "procfs root" + pid path and second one to fit "environ" into it.

How about we add quick helper similar to this:
```rust
pub fn process_file_path<T: AsRef<Path>>(pid: Pid, filename: T) -> PathBuf {
    // It should be max up to 2^22, so 7 bytes should be enough to squeeze string representation of pid?
    static MAX_PID_LENGTH: usize = 7usize;
    let procfs_root = rt::linux::procfs_root();
    let required_len = procfs_root.as_os_str().len() + MAX_PID_LENGTH + filename.as_ref().as_os_str().len();
    let mut result = PathBuf::with_capacity(required_len);
    result.push(procfs_root);
    result.push(pid.to_string());
    result.push(filename.as_ref());

    result
}
```

Reply:
Do you want this to live in `heim_common::utils`?

Reply:
No, it does look like a `heim-process` crate-specific thing, let's put it somewhere into `heim-process/src/sys/linux/*`, your choice.